docs/docs/features/dongle.mdx

```diff
@@ -0,0 +1,349 @@
+For unibody keyboards, the dongle also helps by setting the unibody keyboard as peripheral, optimizing battery usage.
+
+The main benefit of using a dongle is extended and balanced battery life, in some cases extending from a few weeks to several months.
+
+
+![Mermaid Connection](../assets/dongle/mermaid-connection.png)
+
+{/* Here is an example diagram of a split keyboard with a dongle:
+
+```mermaid
+graph TD;
+    A[Computer/Host Device] <-- USB or BLE --> DONGLE[Dongle/Central];
+    DONGLE -. BLE .-> LEFT[Left Half/Peripheral];
+    DONGLE -. BLE .-> RIGHT[Right Half/Peripheral];
+```
+
+Dotted lines represent BLE connections.
+Solid lines represent USB connections. */}
+
+:::info
+For more information on how to set up a unibody or split keyboard, refer to [New Keyboard Shield](../development/new-shield.mdx) under Guides.
+After that is done, you can follow the steps below to add a dongle to the setup.
+:::
+
+## Benefits/Disadvantages of using a Dongle
+
+Benefits:
+
+- For unibody keyboards, the dongle sets the unibody keyboard as peripheral, improving battery life.
+- For split keyboards, adding a dongle enables both keyboard halves to act as peripherals, meaning equal improved battery life for both halves.
+- Easier to connect to the host device.
+
+Disadvantages:
+
+- Extra microcontroller needed
+- Keyboard is unusable without the dongle.
+- Added latency due to the extra hop between the host device and the keyboard halves compared to a USB connected non-dongle setup.
+
+
+
+## Defining a Dongle for a Keyboard 
+
+
+:::note
+In the examples below, we will be referencing the New Keyboard Shield guide files and will be building on top of that.
+Both split and unibody keyboards can be used with a dongle.
+:::
+
+### Keyboard Shield
+
+<Tabs
+defaultValue="split"
+values={[
+{label: 'Split Keyboard', value: 'split'},
+{label: 'Unibody Keyboard', value: 'unibody'},
+]}
+>
+
+<TabItem value="split">
+
+First we will introduce a third split to our keyboard configuration. This will be used as the dongle.
+
+```kconfig title="Kconfig.shield"
+config SHIELD_MY_BOARD_DONGLE
+    def_bool $(shields_list_contains,my_board_dongle)
+
+config SHIELD_MY_BOARD_LEFT
+    def_bool $(shields_list_contains,my_board_left)
+
+config SHIELD_MY_BOARD_RIGHT
+    def_bool $(shields_list_contains,my_board_right)
+```
+</TabItem>
+<TabItem value="unibody">
+We will add a dongle to the unibody keyboard configuration. 
+
+```kconfig title="Kconfig.shield"
+config SHIELD_MY_BOARD_DONGLE
+    def_bool $(shields_list_contains,my_board_dongle)
+
+config SHIELD_MY_BOARD
+    def_bool $(shields_list_contains,my_board)
+
+```
+</TabItem>
+</Tabs>
+
+### Keyboard Defconfig
+
+:::info
+In the shield defconfig we will assume that the user does not want to use the dongle by default.
+This is done so that the user can easily enable the dongle in the user configuration without having to modify the shield configuration.
+:::
```

Comment:
Instead of this note here, I'd write something under the ## header directly stating that this is how to *define* a dongle in such a way that it allows for easy choice between dongle setup and non-dongle setup. Then I'd say we have a separate `## Enabling and Building with a Dongle` with a short text underneath stating that this is how you do it *if the setup is defined flexibly like in the previous section*.

Reply:
restructuring